invenio_app_ils/records/resolver/jsonresolver/document_eitem.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Resolve the EItem referenced in the Document."""
+
+import jsonresolver
+from werkzeug.routing import Rule
+
+from invenio_app_ils.search.api import EItemSearch
+
+from ...api import Document, EItem
+from ..resolver import get_field_value_for_record as get_field_value
+
+# Note: there must be only one resolver per file,
+# otherwise only the last one is registered
+
+
+@jsonresolver.hookimpl
+def jsonresolver_loader(url_map):
+    """Resolve the referred EItems for a Document record."""
+    from flask import current_app
+
+    def eitems_resolver(document_pid):
+        """Search and return the EItems that reference this Document."""
+        eitems = []
+        for hit in EItemSearch().search_by_document_pid(document_pid).scan():
```

Comment:
I think from a data modelling point of view that this is the best approach where each EItem belongs to a document and has a ref to it. But since we mostly work with documents wouldn't storing the EItem pids in the document instead (similar to keywords) perform better since you don't need to scan the index? We'll resolve EItems for a document more often than the other way around, right?

Maybe it's negligible, just curious if this could impact the performance in a negative way.

Reply:
But I think that the documents index will grow much more than the eitem one...

Reply:
Yes, I agree but does looking up specific pids take the same amount of time as scanning?

Reply:
We discussed about this F2F :) Thanks for the comment!